include/boost/geometry/views/closeable_view.hpp

```diff
@@ -44,19 +47,23 @@ struct closing_view
 {
     // Keep this explicit, important for nested views/ranges
     explicit inline closing_view(Range& r)
-        : m_range(r)
+        : m_range(boost::ref(r))
     {}
 
     typedef closing_iterator<Range> iterator;
     typedef closing_iterator<Range const> const_iterator;
 
-    inline const_iterator begin() const { return const_iterator(m_range); }
-    inline const_iterator end() const { return const_iterator(m_range, true); }
+    inline const_iterator begin() const { return const_iterator(cref()); }
+    inline const_iterator end() const { return const_iterator(cref(), true); }
+
+    inline iterator begin() { return iterator(ref()); }
+    inline iterator end() { return iterator(ref(), true); }
 
-    inline iterator begin() { return iterator(m_range); }
-    inline iterator end() { return iterator(m_range, true); }
 private :
-    Range& m_range;
+    inline Range& ref() { return m_range; }
+    inline Range const& cref() const { return m_range; }
+
+    boost::reference_wrapper<Range> m_range;
 };
 
```

Comment:
To be more precise, in order to construct the boundary view of a ring, I take it closeable view (otherwise the boundary view could be missing a segment). As the closeable view depends on the identity view or the closing view. Since I need to copy closeable views, both the closing and identity view need to be copyable.